### ISSUE TITLE
stepper: allow language designers to control printed representation of values

### ISSUE THREAD

**Issue by jbclements** [racket/htdp#56]
This is a (very reasonable) feature request from Mike Sperber. Email pasted (without permission, I'm guessing that's okay):

I've got a minor stepper problem, namely this pair of definitions:

```racket
(define real-make-pair
 (procedure-rename
  (lambda (f r)
    (when (and (not (null? r))
		(not (pair? r)))
      (raise
	(make-exn:fail:contract
	 (string->immutable-string
	  (format "Zweites Argument zu make-pair ist keine Liste, sondern ~e" r))
	 (current-continuation-marks))))
    (cons f r))
  'make-pair))

(define-syntax make-pair
 (let ()
   ;; make it work with match
   (define-struct pair-info ()
     #:super struct:struct-info
     #:property
     prop:procedure
     (lambda (_ stx)
	(syntax-case stx ()
	  ((self . args) (syntax/loc stx (real-make-pair . args)))
	  (else (syntax/loc stx real-make-pair)))))
   (make-pair-info (lambda ()
		      (list #f
			    #'real-make-pair
			    #'pair?
			    (list #'cdr #'car)
			    '(#f #f)
			    #f)))))
```

So make-pair is a macro that looks like the name of a procedure, and
behaves that way in all ways but one - the stepper prints it as
"real-make-pair".  Can I change that somehow?  Help would be much
appreciated!

**Comment by jbclements** [comment]
@mikesperber is this resolved now? (by which I mean: did your enormous amount of work resolve this problem?)

**Comment by mikesperber** [comment]
Ah, oops, I actually resolved this with e02df9ebb71c91f518e959de4f7866cfb6ab810a. Sorry for not noting it here!
